woocommerce/woocommerce-ios
# Make downloadable products more discoverable

Closes: #11387 


## Description
This PR aligns iOS with Android's behavior when dealing with downloadable files:
1. The bottom sheet will show the downloadable files section for simple products and simple subscriptions in two cases: the product is not downloadable or the downloadable files list is empty.
2. The settings rows will show the product downloadable files details when the downloadable files list is not empty.
3. Add an alert to warn the user when they try to make a product non-downloadable. QAkdm5HZdLlG0b8vUkv1pX-fi-5730_229

## Testing instructions
1. Open a simple non-downloadable product.
2. Click on "Add more details"
3. Notice it contains Downloadable files now.
4. Click on Downloadable files.
5. Continue the flow of adding a file.
6. Go back to product details, and confirm the product type reflects "Downloadable" and there is a section for downloadable files.
7. Check in Product settings and confirm the Product Downloadable toggle is checked.
8. Click on Save and confirm it works as expected.
9. Open product settings.
10. Untoggle Product Downloadable.
11. Confirm it shows an alert and that it behaves as expected.

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/1657201/d8922849-c43c-4df8-bb19-dd1c955454b5



---
